x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinder.java

```diff
@@ -529,16 +536,25 @@ static boolean canCreateFromSample(List<String> explanation, String sample, int
                         --fieldsInThisRow;
                     }
 
-                    if (fieldsInLastRow != fieldsInFirstRow) {
-                        explanation.add("Not " + formatName + " because row [" + (numberOfRows - 1) +
-                            "] has a different number of fields to the first row: [" + fieldsInFirstRow + "] and [" +
-                            fieldsInLastRow + "]");
-                        return false;
+                    if (fieldsInThisRow != fieldsInFirstRow) {
+                        illFormattedRows.add(fieldsInFirstRow);
+                        continue;
                     }
 
                     fieldsInLastRow = fieldsInThisRow;
                 }
 
+                // We should only allow a certain percentage of ill formatted rows
+                // as it may effects our sample sizing and down stream processing
+                if (illFormattedRows.size() > (allowedFractionOfBadLines * numberOfRows) ) {
+                    explanation.add(new ParameterizedMessage(
+                        "Not {} because more than {} rows did not have the same number of fields as the first row({}). Bad rows {}",
```

Comment:
```suggestion
                        "Not {} because more than {} rows did not have the same number of fields as the first row ({}). Bad rows {}",
```

Reply:
The message should be adjusted to take account of short-circuiting in the loop above, for example:

```suggestion
                        "Not {} because {} or more rows did not have the same number of fields as the first row ({}). Bad rows {}",
```